hanson.c

```diff
@@ -0,0 +1,312 @@
+// SPDX-License-Identifier: Apache-2.0
+/* Copyright (c) 2025 Elastic NV */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "quark.h"
+
+int	hanson_add_ascii(struct hanson *, int);
+
+static int
+hanson_add(struct hanson *h, void *data, size_t data_len)
+{
+	size_t	r;
+
+	r = fwrite(data, 1, data_len, h->stream);
+	if (unlikely(r != data_len || ferror(h->stream))) {
+		h->error = 1;
+		return (-1);
+	}
+
+	return (0);
+}
+
+/*
+ * Don't change to char, otherwise it will sign extend on wide characters
+ */
+static inline int
+is_escape_char(u_char c)
+{
+	int	v = 0;
+
+	switch (c) {
+	case '\\':	/* FALLTHROUGH */
+	case '\"':	/* FALLTHROUGH */
+	case '\b':	/* FALLTHROUGH */
+	case '\f':	/* FALLTHROUGH */
+	case '\n':	/* FALLTHROUGH */
+	case '\r':	/* FALLTHROUGH */
+	case '\t':	/* FALLTHROUGH */
+		v = 1;
+		break;
+	default:
+		if (c < 32)
+			v = 1;
+		break;
+	}
+
+	return (v);
+}
+
+static int
+hanson_add_string_escaped(struct hanson *h, char *s)
+{
+	int	 r = 0, len, c;
+	char	*p;
+	char	 unicode_buf[16];
+
+	for (p = s; *p != 0; p++) {
+		c = is_escape_char(*p);
+
+		if (likely(!c)) {
+			hanson_add_ascii(h, *p);
+			continue;
+		}
+
+		hanson_add_ascii(h, '\\');
+		switch (c) {
+		case '\\':
```

Comment:
huh, all the cases here are dead code cause `c` is a boolean, right?

Reply:
ouch! yeah, I merged the wrong chunk, is_escape_char used to return the actual char or zero if not escaped, but then I realized that zero must also be escaped, I did test this code, but before commit I merged the wrong chunk.
if should be *p, not c, I'll fix it and add some tests to be sure.

Reply:
zefixed in 479a8650350cec243375455addc433352656d101
I've added tests for all escape sequences as well.